roler/roler.go

```
@@ -116,14 +89,33 @@ func ResolveCredentials(db store.Store, customerID, accessKey, secretKey string)
 
 	arn := aws.StringValue(user.User.Arn)
 	if arn == "" {
-		return creds, fmt.Errorf("No user found when fetching the current user from provided credentials")
+		err = fmt.Errorf("No user found when fetching the current user from provided credentials")
+		logger.WithError(err).Error("error fetching user ARN")
+		return creds, err
 	}
 
 	accountID, err := parseARNAccount(arn)
 	if err != nil {
+		logger.WithError(err).Error("error parsing user ARN")
+		return creds, err
+	}
+
+	// find out if we already have an account saved
+	account, err = db.GetAccount(&store.GetAccountRequest{CustomerID: customerID, Active: true})
+	if err != nil {
+		logger.WithError(err).Error("error reading account from db")
 		return creds, err
 	}
 
+	if account != nil && account.ID != accountID {
+		// yeah, so just delete this sucker and we'll put a new one
+		deleteAccountCredentials(db, account)
```

it might be weird if it gets rolled back, and we end up with a different account than we want. might it be better just to have no account and fail hard?